CONTRIBUTING.md

```diff
@@ -0,0 +1,529 @@
+* an [implementation](#implementation), 
+* a testable [value comparison](#value comparison), and 
+* a list of relevant [citations](#citation-styleguide).  
+
+#### Theory Overview
+
+As the purpose of these tutorials is not to completely derive a method, it is
+better to provide a broad-strokes overview of the formulation of the method, as
+well as explicitly providing (or referencing) all equations to be implemented.
+Fortunately, markdown cells within Jupyter notebooks support LaTeX math formatting,
+and GitHub is capable of rendering these equations when viewing the notebook.
+
+#### Implementation
+
+This section should describe the details of the implementation, with particular
+focus on *why* a certain algorithm or coding decision is made, not just what
+the decision is.  For example, the naive O(N^8) integral transformation within
+MP2 is discussed in the conventional MP2 tutorial
+[here](https://github.com/psi4/psi4numpy/blob/master/Tutorials/05_Moller-Plesset/5a_conventional-mp2.ipynb)
+before the factored O(N^5) transformation is introduced.  This section benefits
+from alternating markdown and code cells, so that the reader can execute the
+cells alongside reading about the implementation.  Again, specific equations
+within publications should be referenced with [LNFA:yy:pp] keys, which should
+match those references within the `Tutorial References` section (see
+[below](#citation-styleguide)).
+
+Note: Every tutorial should be accompanied by a corresponding reference
+implementation.  This is because the interactive tutorials are like a
+classroom, while the reference implementation is like a textbook.  While a more
+adventurous student may skip class to rely on reading the book, every class
+needs an accompanying textbook for supporting material.  The accompanying
+reference implementation can be as simple as reproducing the code cells in a
+single script, or potentially more complex if supplemented by helper
+functions/classes/modules.
+
+### Value Comparison
+
+Each reference implementation and interactive tutorial should conclude with a
+testable value comparison, which makes use of Psi4's `compare_values()`
+function (documented
+[here](http://psicode.org/psi4manual/1.1/api/psi4.driver.compare_values.html?highlight=compare_values#psi4.driver.compare_values)):
+
+```python
+psi4.driver.compare_values(expected, computed, digits, label, exitonfail=True)
+```
+
+where `expected` should be call to Psi4 to compute the desired value (if Psi4
+has such capability) or a hard-coded value (if Psi4 does not have such
+capabilities), and `computed` should be the final output of the reference
+implementation or tutorial.  Value comparison statements will serve an
+important role within [continuous
+integration](#testing-and-continuous-integration), as thse statements are the
+primary means by which the successful execution of a script or tutorial is
+determined.  
+
+## Styleguides
+
+### Git Commit Messages
```

Comment:
This is probably too much. Im not quite sure people will actually use these tags.